modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/util/StreamInterceptorsLoader.java

```diff
@@ -0,0 +1,159 @@
+                }
+            } finally {
+                lock.unlock();
+            }
+        }
+        return Collections.unmodifiableList(interceptors);
+    }
+
+    private static void loadInterceptors() {
+
+        OMElement interceptorsConfig = MiscellaneousUtil.loadXMLConfig(RelayConstants.STREAM_INTERCEPTOR_FILE);
+        if (interceptorsConfig != null) {
+
+            if (!ROOT_Q.equals(interceptorsConfig.getQName())) {
+                handleException("Invalid interceptor configuration file");
+            }
+
+            Iterator iterator = interceptorsConfig.getChildrenWithName(INTERCEPTOR_Q);
+            while (iterator.hasNext()) {
+                OMElement interceptorElem = (OMElement) iterator.next();
+                if (interceptorElem.getAttribute(CLASS_Q) != null) {
+                    String className = interceptorElem.getAttributeValue(CLASS_Q);
+                    if (!"".equals(className)) {
+                        StreamInterceptor interceptor = createInterceptor(className);
+                        interceptors.add(interceptor);
+                        populateParameters(interceptorElem, interceptor);
+                    } else {
+                        handleException("Class name is one or more interceptor");
+                    }
+                } else {
+                    handleException("Class name is one or more interceptor");
+                }
+            }
+        }
+    }
+
+    private static StreamInterceptor createInterceptor(String classFQName) {
+
+        Object obj = null;
+        try {
+            obj = Class.forName(classFQName).newInstance();
+        } catch (ClassNotFoundException | InstantiationException | IllegalAccessException e) {
+            handleException("Error creating Interceptor for class name : " + classFQName, e);
+        }
+        if (obj instanceof StreamInterceptor) {
+            return (StreamInterceptor) obj;
+        } else {
+            handleException(
+                    "Error creating StreamInterceptor. The Interceptor should be of type org.apache.synapse.transport.passthru.StreamInterceptor");
+        }
+        return null;
+    }
+
+    private static void populateParameters(OMElement handlerElem, StreamInterceptor interceptor) {
+
+        for (Iterator it = handlerElem.getChildrenWithName(PARAM_Q); it.hasNext(); ) {
+            OMElement child = (OMElement) it.next();
+
+            String propName = child.getAttribute(NAME_ATT).getAttributeValue();
+            if (propName == null) {
+                handleException("StreamInterceptor parameter must contain the name attribute");
+            } else {
+                if (child.getAttribute(VALUE_ATT) != null) {
+                    String value = child.getAttribute(VALUE_ATT).getAttributeValue();
+                    interceptor.addProperty(propName, value);
+                    PropertyHelper.setInstanceProperty(propName, value, interceptor);
```

Comment:
Will this work? There isn't a method in the Interceptor class to as 'setPropName' right?

Reply:
yes this would, this will find the setter and set this value, same way it works in class mediators.